tests/test_updated_esdl_post_process.py

```diff
@@ -139,10 +139,14 @@ def test_updated_esdl(self):
 
                 # Check pipe diameter
                 if len(fnmatch.filter([energy_system.instance[0].area.asset[ii].id], "Pipe*")) == 1:
-                    if asset_name in ["Pipe1", "Pipe1_ret"]:
+                    if asset_name in ["Pipe1"]:
                         np.testing.assert_array_equal(
                             energy_system.instance[0].area.asset[ii].diameter.name, "DN150"
                         )  # original pipe DN400 being sized
+                    elif asset_name in ["Pipe1_cold_pipe"]:
+                        np.testing.assert_array_equal(
+                            energy_system.instance[0].area.asset[ii].diameter.name, "DN250"
+                        )  # original pipe DN400 being sized
```

Comment:
These 2 pipes now have the same diameter of DN150. Not sure why it is fine now. So I did not change the MIPgap

Reply:
sometimes the order in variables and constraints are created can make the solver try it from a different route and then instead of ending on a MIPgap of 0.05% it will end on 0%.
Since you made Pipe4 optional, the total costs of the system might have gone down, therefore the percental gap larger, thus it has to reduce the size further.

Reply:
@FJanssen-TNO  can this conversation be resolved? Or do you need more info?